test/integration/ggrc/models/factories.py

```diff
@@ -36,9 +36,25 @@ def is_external_custom_attributable(_type):
                     external_customattributable.ExternalCustomAttributable)
 
 
-def random_str(length=8, prefix="", chars=None):
+def random_str(length=8, prefix="", chars=None, suppress_whitespace=True):
+  """Generate random string.
+
+  Args:
+    length: length of expected random string.
+    prefix: string prefix.
+    chars: set of chars for generating.
+    suppress_whitespace: suppress whitespace duplication.
+  Returns:
+    Prefix with random string.
+  """
   chars = chars or string.ascii_uppercase + string.digits + "  _.-"
-  return prefix + "".join(random.choice(chars) for _ in range(length))
+  random_string = ''
+  while len(random_string) < length:
+    random_char = random.choice(chars)
+    if suppress_whitespace and random_string and random_char == " " and random_string[-1] == " ":
```

Comment:
Please consider the following suggestion, I think it might make the code more readable and clear.
```suggestion
is_second_whitespace = bool(
    random_string and random_char == random_string[-1] == " ")
if suppress_whitespace and is_second_whitespace:
  continue
```

Reply:
Fixed. Thank you!